cli/cmd/addon_download.go

```diff
@@ -0,0 +1,180 @@
+			req.Header.Set("Authorization", fmt.Sprintf("Bearer %s", token))
+			return nil
+		}))
+	if err != nil {
+		return fmt.Errorf("failed to create hub client: %w", err)
+	}
+
+	target, err := cmd.Flags().GetString("target")
+	if err != nil {
+		return err
+	}
+
+	addon, addonVersion, err := getAddonMetadata(ctx, c, addonParts[0], addonParts[1], addonVer[0], addonVer[1])
+	if err != nil {
+		return fmt.Errorf("failed to get addon metadata: %w", err)
+	}
+
+	if err := downloadAddon(ctx, c, addon.TeamName, addon.AddonType, addon.Name, addonVersion.Name, addon.AddonFormat, addonVersion.Checksum, target); err != nil {
+		return fmt.Errorf("download failed: %w", err)
+	}
+
+	return nil
+}
+
+func getAddonMetadata(ctx context.Context, c *cloudquery_api.ClientWithResponses, teamName, addonType, addonName, version string) (*cloudquery_api.Addon, *cloudquery_api.AddonVersion, error) {
+	addonResp, err := c.GetAddonWithResponse(ctx, teamName, cloudquery_api.AddonType(addonType), addonName)
+	if err != nil {
+		if addonResp == nil {
+			return nil, nil, fmt.Errorf("failed to get addon: %w", err)
+		}
+		return nil, nil, fmt.Errorf("failed to get addon: %w", errorFromHTTPResponse(addonResp.HTTPResponse, addonResp))
+	}
+
+	addonVersionResp, err := c.GetAddonVersionWithResponse(ctx, teamName, cloudquery_api.AddonType(addonType), addonName, version)
+	if err != nil {
+		if addonVersionResp == nil {
+			return nil, nil, fmt.Errorf("failed to get addon version: %w", err)
+		}
+		return nil, nil, fmt.Errorf("failed to get addon version: %w", errorFromHTTPResponse(addonVersionResp.HTTPResponse, addonVersionResp))
+	}
+	return addonResp.JSON200, addonVersionResp.JSON200, nil
+}
+
+func downloadAddon(ctx context.Context, c *cloudquery_api.ClientWithResponses, teamName string, addonType cloudquery_api.AddonType, addonName, version string, format cloudquery_api.AddonFormat, checksum, targetDir string) (retErr error) {
+	res, err := c.DownloadAddonAsset(ctx, teamName, addonType, addonName, version)
+	if err != nil {
+		return err
+	}
+	if res.StatusCode > 399 {
+		resp, err := cloudquery_api.ParseDownloadAddonAssetResponse(res)
+		if err != nil {
+			return fmt.Errorf("failed to parse %d response: %w", res.StatusCode, err)
+		}
+		return fmt.Errorf("failed to download addon: %w", errorFromHTTPResponse(resp.HTTPResponse, resp))
+	}
+
+	var (
+		fileWriter io.WriteCloser
+		size       int64
+	)
+
+	switch targetDir {
+	case "-":
+		fileWriter = os.Stdout
+	default:
+		filename := strings.Join([]string{teamName, string(addonType), addonName, version}, "_") + "." + string(format)
+		zipPath := filepath.Join(targetDir, filename)
+		f, err := os.Create(zipPath)
+		if err != nil {
+			return fmt.Errorf("failed to create file: %w", err)
+		}
+		fileWriter = f
+
+		defer func() {
+			if retErr != nil {
+				_ = os.Remove(zipPath)
+				return
+			}
+			fmt.Fprintf(os.Stderr, "Wrote %d bytes to %s\n", size, zipPath)
```

Comment:
Closing `os.Stdout` takes care of not writing anything else, but we still want to give some feedback to the user.